src/pages/SignUpPage/index.js

```diff
@@ -19,13 +24,27 @@ class SignUpPage extends React.Component {
   }
 
   handleSubmit = async (values, setSubmitting) => {
+    const { email, firstName, lastName, major, gradYear, password } = values;
     const signupSubmission = {
-      ...values,
+      email,
+      firstName,
+      lastName,
+      major,
+      password,
+      graduationYear: gradYear.toString(),
     };
 
-    await createUserAccountFromSignup(signupSubmission);
+    try {
+      await createNewUser(signupSubmission);
+      await doSignInWithEmailAndPassword(email, password, false);
+      await doSendVerificationEmail();
+      await doSignOut();
 
-    setSubmitting(false);
+      setSubmitting(false);
+    } catch {
+      // TODO: Let user know that an account has already been made with the email they provided
```

Comment:
i think it would be better to isolate the calls into individual try catch blocks so we can see if createNewUser failed, or if verification email failed